Adding Keychron V3, V6, and C3 Support

SparksterZeroS opened the pull request:
I've added Keychron V3 support in the `generate_rules.py` file. I have tested that these are the correct vendor (`VID`) and product (`PID`) ID's.
Thanks to @gassssss and @jn64 for allowing the Keychron C3 and Keychron V6 keyboards to be added as well.

SparksterZeroS committed:
Added Keychron V3 support.

SparksterZeroS committed:
Added support for Keychron C3 and V6

SparksterZeroS committed:
Quick fix to update comment.

The Keychron C3 Pro and Keychron C3 Pro 8k have distinct product IDs. As such, the comment labeling it has been updated to reflect that change.

LeopardEnthusiast commented:
Can you add the Keychron V10 to this PR please?
`Bus 001 Device 002: ID 3434:03a1 Keychron Keychron V10`

SparksterZeroS committed:
Added Keychron V10 support.

G-M0N3Y-2503 commented:
It looks like the Keychron wireless dongle is the same.
```
3434:d030 Keychron  Keychron Link
```

Commit:
Project Reset
tl;dr A lot of refactoring. Python -> Lua. Device IDs in a separate file.

Believe it or not, a lot of these changes were done on a separate repository.
However, I decided not to keep the commit messages as they were really messy and
kind of unprofessional (not like that matters.)

Anyways, here's the rundown.
I changed the way the udev rules were generated. Replacing Python with Lua for no real reason other than I just wanted to learn it and what a better way to do it by trying to write a
somewhat advanced text parser and generator. Device IDs were put into the "devices" files, hence the "text parser" statement from the previous sentence.

denilsonsa commented:
Thanks for you contributions. I've added your commits from your fork into this repository (except the commit(s) that rewrote everything).

Feel free to keep a lua-based approach in your fork. It is indeed a nice opportunity to learn something new.

I personally don't want to modify the structure of this repository that much because it will likely break installation/packaging scripts from multiple distros. Thus, I'd rather keep it conservatively and not break those, unless absolutely needed.
